### PR TITLE
Snoop `test_infra` Names (#1185)

### PR DESCRIPTION
**Pull request by ctodTT**
This change removes the necessity to specify the name of a test in the `compile_and_convert` decorator, using `__name__` of the decorated function to supply this instead. The option to override this choice still exists.

### CODE REVIEW

File: python/test_infra/test_ttir_ops.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+# RUN: %python %s
+
+import inspect
+
+from ttmlir.test_utils import compile_and_convert
+from ttmlir.ttir_builder import Operand, TTIRBuilder
+
+
+@compile_and_convert(((128, 128),))
```
Comment (nsmithtt) [inline python/test_infra/test_ttir_ops.py:13]: Singleton tuples are a gotcha that I'd prefer to avoid, perhaps we should just make this a list to avoid them?  They always result in a weird error message that is totally unrelated to the fix.

File: python/test_infra/test_utils.py [hunk 2]
```diff
@@ -204,86 +190,136 @@ def ttir_to_ttmetal(
...
 
+    print("`ttmetal_to_flatbuffer_file` passed successfully.")
 
-def ttmetal_to_flatbuffer(
-    output_file_name: str = "ttmetal_fb.ttmg",
+
+# ----- Decorators for doing passes and compiling to flatbuffer -----
+
+
+def compile_and_convert(
```
Comment (nsmithtt) [inline python/test_infra/test_utils.py:249]: Maybe we can call this just `compile`?

**Review by nsmithtt — COMMENTED** [review]
This looks great! Minor comments inline.

File: python/test_infra/test_utils.py (hunk 2 above)
Comment (ctodTT) [reply python/test_infra/test_utils.py:249]: That seems a little generic to me, especially if we plan to integrate this into the rest of the LIT tests. I could be wrong here though, LMK what you think.

Comment (nsmithtt) [reply python/test_infra/test_utils.py:249]: How about `compile_to_flatbuffer`?

File: python/test_infra/test_ttir_ops.py (hunk 1 above)
Comment (ctodTT) [reply python/test_infra/test_ttir_ops.py:13]: Also, sorry just noticed, this should be on the other PR (#1236) . These changes won't appear once that one is merged. This applies to both comments

File: python/test_infra/test_utils.py (hunk 2 above)
Comment (ctodTT) [reply python/test_infra/test_utils.py:249]: Oooo I like that much better

File: python/test_infra/test_ttir_ops.py (hunk 1 above)
Comment (nsmithtt) [reply python/test_infra/test_ttir_ops.py:13]: No worries, we can land your other PR first, I accepted that one

**Review by tapspatel — APPROVED** [review]
